src/multipatch/connectivity/edge_transformation.hpp

```diff
@@ -223,29 +291,50 @@ class EdgeTransformation
                 !std::is_same_v<TargetGrid, CurrentGrid>,
                 "The types of the indices should be different");
 
+        using TargetPatch
+                = std::conditional_t<std::is_same_v<TargetGrid, EdgeGrid1>, Patch1, Patch2>;
+
         // Index range of CurrentIdx.
         using IdxRangeCurrent = IdxRange<CurrentGrid>;
         // Index range of other dimension.
         using IdxRangeTarget = IdxRange<TargetGrid>;
         // Coordinate on the discontinuous dimension of the current grid.
         using CurrentCoord = Coord<typename CurrentGrid::continuous_dimension_type>;
-        // Index on the target grid
-        using IdxTarget = typename IdxRangeTarget::discrete_element_type;
+        // Index on the current grid
+        using IdxCurrent = typename IdxRangeCurrent::discrete_element_type;
         // Index step on the target grid
         using IdxStepTarget = typename IdxRangeTarget::discrete_vector_type;
 
+        /*
+            Add a point for the periodic case where the last interpolation point
+            is usually not on the last break point. 
+        */
+        IdxRangeEdge1 const idx_range_patch_1_full(
+                m_idx_range_patch_1.front(),
+                m_idx_range_patch_1.extents() + int(EdgeDim1::PERIODIC));
+        IdxRangeEdge2 const idx_range_patch_2_full(
+                m_idx_range_patch_2.front(),
+                m_idx_range_patch_2.extents() + int(EdgeDim2::PERIODIC));
+
 
         bool is_equivalent_idx = false;
 
         // Get the 1D index range corresponding to the current and target domains.
         IdxRange<EdgeGrid1, EdgeGrid2> const
-                combined_idx_range(m_idx_range_patch_1, m_idx_range_patch_2);
+                combined_idx_range(idx_range_patch_1_full, idx_range_patch_2_full);
         IdxRangeCurrent const current_idx_range(combined_idx_range);
         IdxRangeTarget const target_idx_range(combined_idx_range);
 
         int const n_cells_current = current_idx_range.size() - 1;
         int const n_cells_target = target_idx_range.size() - 1;
 
+        // Periodicity property.
+        if constexpr (CurrentGrid::continuous_dimension_type::PERIODIC) {
+            current_idx = (current_idx - current_idx_range.front()).value() < n_cells_current
+                                  ? current_idx
+                                  : IdxCurrent((current_idx - current_idx_range.back()).value());
```

Comment:
Crappy wifi seems to have lost my comment here. I have 2 comments:
1. Are you sure about the value `IdxCurrent((current_idx - current_idx_range.back()).value())`? I am not sure this works as expected if `current_idx_range.front() != 0`
2. I don't think you need to use integers here. I think it should be (with my correction for point 1 included):
```suggestion
            current_idx = (current_idx < current_idx_range.back()
                                  ? current_idx
                                  : current_idx - current_idx_range.extents());
```

E.g. with `current_idx=11` and `current_idx_range = range(4,10)` we have:
- IdxCurrent((current_idx - current_idx_range.back()).value()) = 11 - 10 = 1
- `current_idx - current_idx_range.extents()` = 11 - 6 = 5

(FYI this is why you are supposed to add/subtract an index step from an index instead of an index as it is harder to make this kind of mistake with steps

Reply:
I didnt know an idx - idxstep could be an idx.

Reply:
> I didnt know an idx - idxstep could be an idx. Or at least I didnt remember.

Hmm, this is kind of the point of an idxstep. It is used to "step" from one index to another. If this was not clear then I guess we need to improve the docs?

Reply:
![image](https://github.com/user-attachments/assets/b675b1f0-9c38-4476-ae61-6aee2ba47ae8)
It looks pretty clear to me. Do you think it is hidden or has it just been a while since you read this part of the docs?

Reply:
I think my mind is just blocked on "a lot of things are not allowed" that I tend to forget what we can do.
It is true, I didnt re-read it for a while. I thought that any operation on Idx will return an IdxStep.

Reply:
I think only Idx-Idx returns an IdxStep (i.e. the distance between the indices)